### PR TITLE
Add multierror package

### PR DESCRIPTION
**Pull request by ronnylt**

### CODE REVIEW

File: multierror/multierror_test.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		formatter := func(es []error) string {
+			s := ""
+			for _, e := range es {
+				s = s + e.Error()
+			}
+			return s
+		}
+
+		err := multierror.Append(e1, e2, e3)
+		res := multierror.Format(err, formatter)
```
Comment (sp-ruben-simon) [inline multierror/multierror_test.go]: missing a test for the default case in Format()

File: multierror/multierror.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package multierror
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Append appends error elements to the end of a error slice, returning a new error
```
Comment (dpinol) [inline multierror/multierror.go]: "a error" -> "an error"

File: multierror/multierror_test.go (hunk 1 above)
Comment (ronnylt) [reply multierror/multierror_test.go]: Fixed

File: multierror/multierror.go (hunk 2 above)
Comment (ronnylt) [reply multierror/multierror.go]: Fixed

**Comment by gonzaloserrano** [comment]
Don't you like https://github.com/pkg/errors#adding-context-to-an-error ? :-)

**Comment by dpinol** [comment]
@gonzaloserrano Wrap is great, but here we wanted to gather a collection of errors in a single one

**Pull request merged**